lms/djangoapps/courseware/tests/test_module_render.py

```diff
@@ -113,6 +114,25 @@ def set_score(self, json_data, suffix):  # pylint: disable=unused-argument
         )
 
 
+class CompletableStatelessXBlock(XBlock):
+    """
+    This XBlock exists to test grade storage for blocks that don't store
+    student state in a scoped field.
+    """
+
+    @XBlock.json_handler
+    def complete(self, json_data, suffix):  # pylint: disable=unused-argument
+        self.runtime.publish(
+            self,
+            'completion',
+            {'completion': json_data['completion']},
+        )
+
+    @XBlock.json_handler
+    def progress(self, json_data, suffix):  # pylint: disable=unused-argument
```

Comment:
I see this is test code, but maybe add something like?:
```
""" Progress is deprecated.  New code should use complete. """
```
I would add a similar note to almost any use of progress to both clarify what it is and remind people not to continue with it.